contracts/PositionManagerWETH.sol

```diff
@@ -34,6 +34,10 @@ contract PositionManagerWETH is IPositionManagerWETH, PositionManagerWrappedColl
         payable
         override
     {
+        if (collateralChange == 0 && msg.value > 0) {
```

Comment:
I think this is the correct check
```suggestion
        if (!isCollateralIncrease && msg.value > 0) {
```

Reply:
I think we need a combination of both since neither of these covers the case where `isCollateralIncrease = true` and `collateralChange == 0`.

```suggestion
        if ((collateralChange == 0 || !isCollateralIncrease) && msg.value > 0) {
```

Reply:
but at this point I think we can also consider moving the check further down the function that emits `CollateralChangeAmountDoesNotMatchETHValue` and just validate all the invalid msg.value combos in one place and emit a single generic error (`MsgValueMismatch` or something)